### ISSUE TITLE
fieldWysiwyg shows incorrect content

### ISSUE THREAD

**Issue by mpelzsherman** [department-of-veterans-affairs/va.gov-team#15321]
The content in some fieldContentBlock.fieldWysiwyg in the cms export differs from production.
Note the data structure and entity ID's are the same, but the content is different.
This _seems_ to point to a problem with the export.

e.g.: 

### GraphQL:

```json
    "fieldContentBlock": [
        {
            "entity": {
                "entityType": "paragraph",
                "entityBundle": "wysiwyg",
                "entityId": "5331",
                "fieldWysiwyg": {
                    "processed": "<html><head></head><body><h2>Requesting chaplain services</h2>\n\n<p>Chaplains are available 24 hours a day, ..."
                }
            }
        }
    ]
```

### CMS export (UUID node.1d014daa-80cd-46f6-952d-ad1b488a561a.json)

```
    "fieldContentBlock": [
        {
            "contentModelType": "paragraph-wysiwyg",
            "entity": {
                "entityType": "paragraph",
                "entityBundle": "wysiwyg",
                "fieldWysiwyg": {
                    "processed": "<html><head></head><body><h2>Spiritual and religious services</h2>\n\n<p>Our chaplains help patients and family members through&#xA0;outpatient treatment, hospital stays, and extended care. If needed, they will contact your personal clergy or spiritual care provider in the community...."
                },
                "entityId": "5331"
            },
            "entityType": "paragraph",
            "entityBundle": "wysiwyg",
            "entityId": "5331"
        }
    ],
```
### GraphQL:
```
"fieldContentBlock": [
--
  | {
  | "entity": {
  | "entityType": "paragraph",
  | "entityBundle": "wysiwyg",
  | "entityId": "3784",
  | "fieldWysiwyg": {
  | "processed": "...href=\"/pittsburgh-health-care/medical-records-office\" title=\"Medical records office\">Access your health records</a>..."
  | }
  | }
  | },
```
### CMS Export (node.4c194aaa-a67f-4212-993f-7e55c0f14ccf.json):

```
"fieldContentBlock": [
--
  | {
  | "contentModelType": "paragraph-wysiwyg",
  | "entity": {
  | "entityType": "paragraph",
  | "entityBundle": "wysiwyg",
  | "fieldWysiwyg": {
  | "processed": "... href=\"/pittsburgh-health-care/medical-records-office\" title=\"Medical records office\">Medical records office</a>..."
  | },
  | "entityId": "3784"
  | },

**Comment by indytechcook** [comment]
I found the root cause but not a solution yet.

The `revision_id` being referenced in the field is different than the `default` revision for the `paragraph`.

The UUID is associated with the `default` revision for the paragraph. So when `tome_sync` loads the data for the UUID it loads the incorrect version.

Field `field_content_block`
```            [field_content_block] => Array
                (
                    [x-default] => Array
                        (
                            [0] => Array
                                (
                                    [target_id] => 5331
                                    [target_revision_id] => 35411
                                )

                        )

                )
```

Data from the paragraph tables shows the default paragraph is different from than the default
```
SELECT t.*
FROM drupal8.paragraphs_item_field_data t
WHERE id=5331
```

```
| id   | revision_id | type    | langcode | status | created    | parent_id | parent_type | parent_field_name   | behavior_settings | default_langcode | revision_translation_affected |
|------|-------------|---------|----------|--------|------------|-----------|-------------|---------------------|-------------------|------------------|-------------------------------|
| 5331 | 147531      | wysiwyg | en       | 1      | 1565710023 | 8322      | node        | field_content_block | a:0:{}            | 1                | 1                             |

```

Looking at the revisions for this paragraph, there are several `revision_default` rows and there should only be one.

```
SELECT t.*
FROM drupal8.paragraphs_item_revision t
WHERE id=5331
```

Truncated list
```
| id   | revision_id | langcode | revision_default |
|------|-------------|----------|------------------|
| 5331 | 23903       | en       | 1                |
| 5331 | 23908       | en       | 1                |
| 5331 | 23910       | en       | 1                |
| 5331 | 24567       | en       | 1                |
| 5331 | 24582       | en       | 1                |
| 5331 | 24664       | en       | 1                |
| 5331 | 25631       | en       | 1                |
| 5331 | 25644       | en       | 1                |
| 5331 | 25647       | en       | 1                |
| 5331 | 25649       | en       | 1                |
| 5331 | 25666       | en       | 1                |
| 5331 | 26751       | en       | 0                |
| 5331 | 26753       | en       | 0                |
| 5331 | 26755       | en       | 0                |
```

So the field is looking at revision id `35411` and tome_sync is looking for `147531` since tome_sync only looks at the uuid.

My current direction is looking at why the `paragraphs_item_revision` has the incorrect `revision_id`

**Comment by indytechcook** [comment]
Current Follow Issue:

Quick fix for tome_syc export: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/3325
Address the root cause: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/3324

**Comment by mpelzsherman** [comment]
Blocked by https://github.com/department-of-veterans-affairs/va.gov-cms/issues/3324

**Comment by indytechcook** [comment]
Please reset, https://github.com/department-of-veterans-affairs/va.gov-cms/issues/3324 has been merged and content fix script has been run.

**Comment by mpelzsherman** [comment]
Sorry @indytechcook but it appears the content for https://prod.cms.va.gov/pittsburgh-health-care/chaplain-services-and-interfaith-chapels is still not displaying correcly in the CMS Export build.

I'm getting the same result for that node as I posted in the original ticket.

pittsburgh-health-care/contact-us/index.html looks good though.

**Comment by mpelzsherman** [comment]
The CMS team is [still working on fixing data export issues](https://dsva.slack.com/archives/C01512KN35G/p1605026163157500?thread_ts=1604955651.153200&cid=C01512KN35G). Moving the ticket back to "Blocked".

**Comment by indytechcook** [comment]
New issue created:

* FIxes what causes corrupt data: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/3449
* Fixes the corrupted data: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/3450

**Comment by mpelzsherman** [comment]
Per @indytechcook there is another (undocumented) [issue](https://dsva.slack.com/archives/C01512KN35G/p1606314533315600?thread_ts=1606253582.310900&cid=C01512KN35G) affecting the CMS data. Moving this back to blocked.

**Comment by indytechcook** [comment]
@mpelzsherman this should unblocked now.  Again!

**Comment by mpelzsherman** [comment]
I just compared the files using the latest export, and there are 0 differences! Thanks @indytechcook !
